src/index.ts

```diff
@@ -33,19 +34,18 @@ export function makeTrustedForwarderMulticall(transactions: TransactionRequest[]
   }
 }
 
-export async function callWithOffchainData(
+export async function callWithOffchainData (
   transactions: [TransactionRequest, ...TransactionRequest[]],
-  provider: Parameters<typeof viem.custom>[0],
+  client: viem.PublicClient,
   adapters: OracleAdapter[]
 ): Promise<[viem.Hex, ...viem.Hex[]]> {
-  const client = viem.createPublicClient({ transport: viem.custom(provider, { retryCount: 0 }) })
```

Comment:
Does this line not work? It's how we enable support for ethers and other clients.